lib/shoulda/matchers/active_record/have_secure_token_matcher.rb

```diff
@@ -110,7 +110,7 @@ def has_expected_instance_methods?
         end
 
         def has_expected_db_column?
-          matcher = HaveDbColumnMatcher.new(token_attribute).of_type(:string)
+          matcher = HaveDbColumnMatcher.new(token_attribute).of_type(:string) || HaveDbColumnMatcher.new(token_attribute).of_type(:text)
```

Comment:
This is always going to return the first one, the one for `:string`. The reason is that this line merely returns a matcher object; it doesn't actually perform the matching. That happens in the next line. 

What you probably want to do is:

``` ruby
matchers = [
  HaveDbColumnMatcher.new(token_attribute).of_type(:string),
  HaveDbColumnMatcher.new(token_attribute).of_type(:text)
]
matchers.any? { |matcher| matcher.matches?(@subject) }
```